Jenkinsfile.groovy

```diff
@@ -1,130 +1,202 @@
-		}
-		failure {
-			setBuildStatus("Build failed", "FAILURE", "${STATUS_CONTEXT}", "${GITHUB_BRANCH_HEAD_SHA}")
-		}
-	}
+    agent any
+    environment {
+        def dllsToDeployX64 = 'CookComputing.XmlRpcV2 MetriCam2.Cameras.ifm MetriCam2.Cameras.Kinect2 MetriCam2.Cameras.OrbbecOpenNI MetriCam2.Cameras.Sick.TiM561 MetriCam2.Cameras.Sick.VisionaryT MetriCam2.Cameras.SVS MetriCam2.Cameras.UEye MetriCam2.Cameras.WebCam MetriCam2.Cameras.BaslerToF'
+        def dllsToDeployAnyCPU = 'MetriCam2.Controls MetriCam2 Metrilus.Util Newtonsoft.Json MetriCam2.Cameras.RealSense2'
+        def dllsToDeployNetStandard = 'MetriCam2.NetStandard Metrilus.Util.NetStandard MetriCam2.Cameras.RealSense2.NetStandard'
+
+        def currentBranch = "${env.GITHUB_BRANCH_NAME}"
+        def msbuildToolName = 'MSBuild Release/x64 [v15.0 / VS2017]'
+        def solutionFilename = 'MetriCam2_SDK.sln'
+
+        def releaseVersion = getReleaseVersion(currentBranch);
+        def releaseFolder = getReleaseFolder(currentBranch, releaseVersion)
+
+        def releaseDirectory = "Z:\\\\releases\\MetriCam2\\${releaseFolder}"
+        def releaseLibraryDirectory = "${releaseDirectory}\\lib"
+        def releaseSuffixNetStandard20 = "_netstandard2.0"
+        def releaseSuffixDebug = "_debug"
+
+        def STATUS_CONTEXT = 'MetriCam2 CI'
+        def BUILD_DATETIME = new Date(currentBuild.startTimeInMillis).format("yyyyMMdd-HHmm")
+        def DEBUG_SUFFIX = "_debug"
+    }
+    stages {
+        stage('Pre-Build') {
+            steps {
+                echo "Set build status: pending"
+                setBuildStatus("Build started", "PENDING", "${STATUS_CONTEXT}", "${GITHUB_BRANCH_HEAD_SHA}")
+
+                bat '''
+                    @echo Restoring NuGet Packages ...
+                    %NUGET_EXE% restore
+                    '''
+
+                bat '''
+                    @echo Checking References ...
+                    %REFERENCE_CHECKER% -j "%JOB_NAME%" "%WORKSPACE%"
+                    '''
+
+                bat """
+                    @echo Updating Assembly-Info ...
+                    \"Scripts\\Set Assembly-Info Version.cmd\" \"SolutionAssemblyInfo.cs\" ${releaseVersion}
+                    """
+            }
+        }
+
+        stage('Build') {
+            steps {
+                bat "\"${tool msbuildToolName}\" ${solutionFilename} /p:Configuration=Release;Platform=x64"
+                bat "\"${tool msbuildToolName}\" ${solutionFilename} /p:Configuration=Debug;Platform=x64"
+            }
+        }
+
+        stage('Publish') {
+            environment {
+                def RELEASE_DIR_X64 = 'bin\\x64\\Release\\'
+                def DEBUG_DIR_X64 = 'bin\\x64\\Debug\\'
+                def RELEASE_DIR_ANYCPU = 'bin\\Release\\'
+                def DEBUG_DIR_ANYCPU = 'bin\\Debug\\'
+                def RELEASE_DIR_NETSTANDARD = 'bin\\Release\\netstandard2.0\\'
```

Comment:
Warum nicht so etwas wie `def RELEASE_DIR_NETSTANDARD = RELEASE_DIR_ANYCPU + 'netstandard2.0\\'` ?

Reply:
Joah. Fänd ich schön, wenn es bei allen klappen würd. Aber nachdem x64 eh extra definiert werden muss seh ich keinen großen Vorteil.

Reply:
Geht imo mehr um die lesbarkeit: netstandard kommt in nen Unterordner von AnyCPU. Ist aber Geschmackssache.